vcx/libvcx/src/utils/libindy/ledger.rs

```diff
@@ -84,3 +90,359 @@ pub fn libindy_build_create_credential_def_txn(submitter_did: &str,
+    -> Result<(String, String, String, Option<PaymentTxn>), u32> {
+    debug!("creating revocation registry definition with issuer_did: {}, cred_def_id: {}, tails_file_path: {}, max_creds: {}",
+           issuer_did, cred_def_id, tails_file, max_creds);
+    if settings::test_indy_mode_enabled() { return Ok((REV_REG_ID.to_string(), REV_DEF_JSON.to_string(), "".to_string(), None)); }
+
+    let (rev_reg_id, rev_reg_def_json, rev_reg_entry_json) = libindy_create_and_store_revoc_reg(
+        issuer_did,
+        cred_def_id,
+        tails_file,
+        max_creds
+    )?;
+
+    let rev_reg_def_req = libindy_build_revoc_reg_def_request(issuer_did, &rev_reg_def_json)?;
+
+    let (payment, _) = pay_for_txn(&rev_reg_def_req, REV_REG_DEF_TXN_TYPE)?;
+
+    Ok((rev_reg_id, rev_reg_def_json, rev_reg_entry_json, payment))
+}
+
+pub fn get_rev_reg_def_json(rev_reg_id: &str) -> Result<(String, String), u32> {
+    if settings::test_indy_mode_enabled() { return Ok((REV_REG_ID.to_string(), REV_DEF_JSON.to_string())); }
+
+    let submitter_did = settings::get_config_value(settings::CONFIG_INSTITUTION_DID)?;
+
+    libindy_build_get_revoc_reg_def_request(&submitter_did, rev_reg_id)
+        .and_then(|req| libindy_submit_request(&req))
+        .and_then(|response| libindy_parse_get_revoc_reg_def_response(&response))
+}
+
+pub fn post_rev_reg_delta(issuer_did: &str, rev_reg_id: &str, rev_reg_entry_json: &str)
+                          -> Result<(Option<PaymentTxn>, String), u32> {
+    libindy_build_revoc_reg_entry_request(issuer_did, rev_reg_id, REVOC_REG_TYPE, rev_reg_entry_json)
+        .and_then(|req| pay_for_txn(&req, REV_REG_DELTA_TXN_TYPE))
+}
+
+pub fn get_rev_reg_delta_json(rev_reg_id: &str, from: Option<u64>, to: Option<u64>)
+    -> Result<(String, String, u64), u32> {
+    if settings::test_indy_mode_enabled() { return Ok((REV_REG_ID.to_string(), REV_REG_DELTA_JSON.to_string(), 1)); }
+
+    let submitter_did = settings::get_config_value(settings::CONFIG_INSTITUTION_DID)?;
+    let from: i64 = if let Some(_from) = from { _from as i64 } else { -1 };
+    let to = if let Some(_to) = to { _to as i64 } else { time::get_time().sec };
+
+    libindy_build_get_revoc_reg_delta_request(&submitter_did, rev_reg_id, from, to)
+        .and_then(|req| libindy_submit_request(&req))
+        .and_then(|response| libindy_parse_get_revoc_reg_delta_response(&response))
+}
+
+pub fn get_rev_reg(rev_reg_id: &str, timestamp: u64) -> Result<(String, String, u64), u32> {
+    if settings::test_indy_mode_enabled() { return Ok((REV_REG_ID.to_string(), REV_REG_JSON.to_string(), 1)); }
+    let submitter_did = settings::get_config_value(settings::CONFIG_INSTITUTION_DID)?;
+
+    libindy_build_get_revoc_reg_request(&submitter_did, rev_reg_id, timestamp)
+        .and_then(|req| libindy_submit_request(&req))
+        .and_then(|response| libindy_parse_get_revoc_reg_response(&response))
+}
+
+pub fn revoke_credential(tails_file: &str, rev_reg_id: &str, cred_rev_id: &str)
```

Comment:
@glowkey @dkulic I don't like the idea of mixing anoncreds and ledger logic inside `utils/libindy/ledger.rs` but if it's common approach in VCX for now - we can accept it and don't perform global refactoring in scope of this PR

Reply:
I agree.  There is a separate anoncreds.rs module that can be used.

Reply:
Is this blocking the ticket to be merged?

Reply:
I think so

Reply:
OK, so libvcx should be refactored in this ticket also? :)

Reply:
@dkulic As far as I understood @glowkey - such kind of mixing is not a style for VCX. So please organize new code similar as it's done in existing codebase.

Reply:
@ryMarsh44 @glowkey I am not sure I understand fully how this should be reorganized, may you help me with that?